nodejs/aws-infra/experimental/clusterAutoScaling.ts

```diff
@@ -0,0 +1,566 @@
+    subnetIds: pulumi.Input<string>[],
+    parameters: pulumi.Input<TemplateParameters>): pulumi.Output<string> {
+
+    const subnetIdsArray = pulumi.all(subnetIds);
+    return pulumi.all([subnetIdsArray, instanceLaunchConfigurationId, parameters])
+                 .apply(([subnetIdsArray, instanceLaunchConfigurationId, parameters]) => {
+
+    const minSize = parameters.minSize !== undefined ? parameters.minSize : 2;
+    const maxSize = parameters.maxSize !== undefined ? parameters.maxSize : 100;
+
+    return `
+    AWSTemplateFormatVersion: '2010-09-09'
+    Outputs:
+        Instances:
+            Value: !Ref Instances
+    Resources:
+        Instances:
+            Type: AWS::AutoScaling::AutoScalingGroup
+            Properties:
+                Cooldown: 300
+                DesiredCapacity: ${minSize}
+                HealthCheckGracePeriod: 120
+                HealthCheckType: EC2
+                LaunchConfigurationName: "${instanceLaunchConfigurationId}"
+                MaxSize: ${maxSize}
+                MetricsCollection:
+                -   Granularity: 1Minute
+                MinSize: ${minSize}
+                VPCZoneIdentifier: ${JSON.stringify(subnetIdsArray)}
+                Tags:
+                -   Key: Name
+                    Value: ${instanceName}
+                    PropagateAtLaunch: true
+            CreationPolicy:
+                ResourceSignal:
+                    Count: ${minSize}
+                    Timeout: PT15M
+            UpdatePolicy:
+                AutoScalingRollingUpdate:
+                    MaxBatchSize: 1
+                    MinInstancesInService: ${minSize}
+                    PauseTime: PT15M
+                    SuspendProcesses:
+                    -   ScheduledActions
+                    WaitOnResourceSignals: true
+    `;
+                 });
+}
+
+export interface ClusterAutoScalingGroupArgs {
+    /**
+     * Cluster to create the autoscaling group for.
+     */
+    cluster: mod.Cluster;
+
+    /**
+     * The config to use when creating the auto scaling group.
+     *
+     * [launchConfiguration] or [launchConfigurationArgs] can be provided.  And, if either are
+     * provided will be used as the launch configuration for the auto scaling group.
+     *
+     * If neither are provided, a default instance will be create by calling
+     * [cluster.createAutoScalingConfig()].
+     */
+    launchConfiguration?: ClusterAutoScalingLaunchConfiguration;
+
+    /**
+     * The config to use when creating the auto scaling group.
+     *
+     * [launchConfiguration] or [launchConfigurationArgs] can be provided.  And, if either are
+     * provided will be used as the launch configuration for the auto scaling group.
+     *
+     * If neither are provided, a default instance will be create by calling
+     * [cluster.createAutoScalingConfig()].
+     */
+    launchConfigurationArgs?: ClusterAutoScalingLaunchConfigurationArgs;
```

Comment:
see above for discussion on why we supply launchconfiguration and launchConfigurationArgs.